test/unit/int_algolia/scripts/algolia/model/algoliaLocalizedContent.test.js

```diff
@@ -0,0 +1,134 @@
+'use strict';
+
+// Import the module and mocks
+var AlgoliaLocalizedContent = require('../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/model/algoliaLocalizedContent');
+var ContentMock = require('../../../../../mocks/dw/content/Content');
+var GlobalMock = require('../../../../../mocks/global');
+var ObjectHelper = require('../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/helper/objectHelper');
+// Mock global dependencies
+global.request = new GlobalMock.RequestMock();
+
+// Jest mocks for imported modules
+jest.mock('dw/web/URLUtils', () => {
+    return {
+        url: function(endpoint, param, id) {
+            var relURL = '/on/demandware.store/Sites-Algolia_SFRA-Site/';
+            return relURL + global.request.getLocale() + '/' + endpoint + '?' + param + '=' + id;
+        }
+    }
+}, {virtual: true});
+
+jest.mock('*/cartridge/scripts/algolia/lib/pageDesignerHelper', () => {
+    return jest.requireActual('../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/lib/pageDesignerHelper');
+}, {virtual: true});
+
+jest.mock('*/cartridge/scripts/algolia/helper/objectHelper', () => {
+    return jest.requireActual('../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/helper/objectHelper');
+}, {virtual: true});
+
+jest.mock('*/cartridge/scripts/algolia/lib/algoliaContentConfig', () => {
+    return jest.requireActual('../../../../../../cartridges/int_algolia/cartridge/scripts/algolia/lib/algoliaContentConfig');
+}, {virtual: true});
+
+jest.mock('dw/experience/PageMgr', () => {
```

Comment:
Since you use it in multiple tests, you can maybe mock in once in `jest.setup.js`